crates/gpui/examples/window.rs

```diff
@@ -0,0 +1,201 @@
+        WindowBounds::Windowed(Bounds::centered(None, size(px(300.0), px(300.0)), cx))
+    }
+
+    fn new_normal_window(cx: &mut AppContext) {
+        let window_bounds = Self::window_bounds(cx);
+
+        cx.open_window(
+            WindowOptions {
+                window_bounds: Some(window_bounds),
+                ..Default::default()
+            },
+            |cx| {
+                cx.new_view(|_cx| SubWindow {
+                    custom_titlebar: false,
+                })
+            },
+        )
+        .unwrap();
+    }
+
+    fn new_popup_window(cx: &mut AppContext) {
+        let window_bounds = Self::window_bounds(cx);
+
+        cx.open_window(
+            WindowOptions {
+                window_bounds: Some(window_bounds),
+                kind: WindowKind::PopUp,
+                ..Default::default()
+            },
+            |cx| {
+                cx.new_view(|_cx| SubWindow {
+                    custom_titlebar: false,
+                })
+            },
+        )
+        .unwrap();
+    }
+
+    fn new_custom_titlebar_window(cx: &mut AppContext) {
+        let window_bounds = Self::window_bounds(cx);
+
+        cx.open_window(
+            WindowOptions {
+                titlebar: None,
+                window_bounds: Some(window_bounds),
+                ..Default::default()
+            },
+            |cx| {
+                cx.new_view(|_cx| SubWindow {
+                    custom_titlebar: true,
+                })
+            },
+        )
+        .unwrap();
+    }
+
+    fn new_hide_window(cx: &mut AppContext) {
+        let window_bounds = Self::window_bounds(cx);
+
+        cx.open_window(
+            WindowOptions {
+                show: false,
+                window_bounds: Some(window_bounds),
+                ..Default::default()
+            },
+            |cx| {
+                cx.new_view(|_cx| SubWindow {
+                    custom_titlebar: false,
+                })
+            },
+        )
+        .unwrap();
+    }
+
+    fn new_unmovable_window(cx: &mut AppContext) {
+        let window_bounds = Self::window_bounds(cx);
+
+        cx.open_window(
+            WindowOptions {
+                is_movable: false,
+                titlebar: None,
+                window_bounds: Some(window_bounds),
+                ..Default::default()
+            },
+            |cx| {
+                cx.new_view(|_cx| SubWindow {
+                    custom_titlebar: false,
+                })
+            },
+        )
+        .unwrap();
+    }
+}
+
+impl Render for WindowDemo {
+    fn render(&mut self, _cx: &mut ViewContext<Self>) -> impl IntoElement {
+        div()
+            .p_4()
+            .flex()
+            .flex_wrap()
+            .bg(rgb(0xffffff))
+            .size_full()
+            .justify_center()
+            .items_center()
+            .gap_2()
+            .child(button("Normal", |cx| {
+                Self::new_normal_window(cx);
```

Comment:
Can we inline all of these functions? I don't think we need the extra indirection here :)